.github/workflows/setup.yml

```diff
@@ -1,41 +1,63 @@
 # This is a basic workflow to help you get started with Actions
 
-name: CI
+name: Setup
 
-# Controls when the workflow will run
 on:
   # Triggers the workflow on push or pull request events but only for the "main" branch
   push:
-    branches: [ "main" ]
-  pull_request:
-    branches: [ "main" ]
+    branches: [ "open_pdks_testing" ]
+  # pull_request:
 
-  # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
   build:
-    # The type of runner that the job will run on
     runs-on: ubuntu-latest
 
-    # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v4
-
+      - uses: actions/checkout@v4 
+      # Downloads all dependencies if they don't exist
       # Runs commands to install xschem, then the PDK files, then open xschem
-        # First steps ensure all dependencies are downloaded
-          # Automatically says yes to all installs (hopefully)
-      - name: Download Tools
+      
+      - name: Install Dependencies 
+        run: |
+              sudo rm -rf /usr/share/dotnet
+              sudo rm -rf "$AGENT_TOOLSDIRECTORY"
+              sudo apt-get update
+              sudo apt-get install -y tk8.6-dev libxpm-dev tcl8.6-dev libx11-dev flex libtool-bin libxaw7-dev libreadline6-dev libcairo2-dev
+      - name: Install magic
+        run: | 
+              git clone --depth 1 https://github.com/RTimothyEdwards/magic
+              cd magic
+              ./configure
+              make database/database.h
+              make -j$(nproc)
+              sudo apt-get install magic
+      - name: Install ngspice
+        run: |
+              git clone --depth 1 git://git.code.sf.net/p/ngspice/ngspice
+              cd ngspice
+              mkdir release
+              ./autogen.sh
+              cd release
+              ../configure --with-x --enable-xspice --disable-debug --enable-cider --with-readline=yes --enable-openmp --enable-osdi
+              make
+      - name: Install skywater-pdk Libraries     
+        run: |
+              git clone https://github.com/google/skywater-pdk.git
+              cd skywater-pdk
+              SUBMODULE_VERSION=latest make submodules -j3 || make submodules -j1
+              make timing
+      - name: Generate PDK files
         run: |
-              echo Hello, world!
-              sudo apt-get install tk8.6-dev libxpm-dev tcl8.6-dev libtk-dev libx11-dev -y
+              git clone --depth 1 git://opencircuitdesign.com/open_pdks
+              cd open_pdks
+              sudo ./configure --with-sky130-source=~/skywater-pdk/libraries --with-sky130-local-path=~/cad/pdks
+              make
+              sudo make install
```

Comment:
You shouldn't do `sudo`. Instead, set the prefix to a local directory

Reply:
I'm not sure I understand? Do you mean on Line 55?